exercises/rectangles/example.go

```diff
@@ -0,0 +1,48 @@
+package rectangles
+
+import "strings"
+
+const ignore = -1
+
+func checkSide(diagram []string, start, end, rowIndex, colIndex int, pattern string) bool {
+	var prop string
+	for i := start; i < end; i++ {
+		if rowIndex == ignore {
+			prop = diagram[i][colIndex : colIndex+1]
+		} else {
+			prop = diagram[rowIndex][i : i+1]
+		}
+		if !strings.Contains(pattern, prop) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
I'm thinking that checkSide might be split into two functions, neither of which requires a call with an `ignore` parameter, since it is constant within the function to act like a control variable for path selection. It would make the code longer but perhaps more readable.  What do you think ?  Seems that one case is horizontal and one is vertical side checking, if I am understanding the code accurately.  The pattern might also be local to both the replacement functions as well.

Reply:
I think this is a great idea since I believe readability is the utmost important aspect of example. Done so.